src/main/java/com/undertheriver/sgsg/foler/domain/Folder.java

```diff
@@ -53,6 +53,14 @@
     
     private Boolean secret;
 
+    public boolean isSecret() {
+        if (secret == null) {
+            return false;
+        }
+
+        return secret;
+    }
+
```

Comment:
리뷰가 반영되면 이 로직도 제거될 수 있겠네요

Reply:
secret이 Boolean 타입이라 null일 수도 있는데요.
만약 null을 리턴하게 되면 if문에서 에러가 발생해서 null 체크 후 false로 변경해서 리턴하고 있습니다.
그래서 리뷰 반영 후에도 필요할 것 같습니다!!

Reply:
코멘트의 설명이 부실한 것 같아서 설명을 추가적으로 하겠습니다!!
만약 `isSecret()` 메서드가 없다면 `getSecret()` 메서드를 호출해서 체크를 해야 할 것 같은데요.
아래와 같이요.

```java
public String fetchContent() {
   if (folder.getSecret()) {  // 여기가 if (null) 이 될 수도 있음.
                return EMPTY_STRING;
   }
   return this.content;
}
```

`secret`이 `Boolean` 타입이다 보니 `null, false, true` 세 가지의 값을 가질 수 있더라구요.
그런데 테스트해보니 `if (null)` 는 NPE 발생시키더라구요.
이런 이유로 리뷰가 반영 되어도 이 로직이 필요할 것 같습니다.

이 로직을 작성할 때 고려한 방법이 두 가지인데요.

1. `getSecret()`을 true 와 false만 리턴하도록 작성 (null 체크 후 false로 치환)
2. `isSecret()` 구현

두 가지의 방법 중 저는 2번을 택했습니다.

Reply:
흠 꼭 필요하군요..! 제가 잘몰랐습니다 😥
2번이 좋겠네요..! 꺼내서 쓰는것보다 메시지를 보내는형태를 선호합니다

```suggestion
    public boolean isSecret() {
        return Objects.equals(secret, Boolean.TRUE)
    }
```
이렇게 바꿀수도있겠네요

혹은 return secret ==  Boolean.TRUE